Sequence class (>2.1.4) adds characters which are not accepted by codecs

![image](https://user-images.githubusercontent.com/20915439/121111868-f8aea280-c852-11eb-81b6-e633ea173053.png)

When using the render_dot_tree function under py_trees.display, I encountered an issue whereby the writer could not successfully parse Sequence classes. I created a graph with a simple Behaviour Tree for rendering and it continuously failed unless reverted back to version 2.1.4 of py_trees. See the error below.

To reproduce:

Setup a non utf-8 locale
```
$ cat /usr/share/i18n/SUPPORTED | grep en_US
$ sudo locale-gen en_US.ISO-8859-15
$ sudo update-locale
$  locale -a
```

Confirm that unicode is not available:
```
$ python3 ./console.py
dude
dude
dude
dude
dude
dude
dude
dude
dude
dude
dude
dude
dude
dude
dude
dude
dude
dude
dude
dude
[DEBUG] loginfo message
[ WARN] logwarn message
[ERROR] logerror message
[FATAL] logfatal message
red
some normal text
    Name: Dude
special characters are

Traceback (most recent call last):
  File "./console.py", line 343, in <module>
    print("lightning_bolt: {}".format(lightning_bolt))
  File "/usr/lib/python3.8/encodings/iso8859_15.py", line 19, in encode
    return codecs.charmap_encode(input,self.errors,encoding_table)[0]
UnicodeEncodeError: 'charmap' codec can't encode character '\u26a1' in position 16: character maps to <undefined>
```

Then call the render demo to tickle the error.
```
py-trees-demo-dot-graphs 

*******************************************************************************
                                   Dot Graphs                                  
*******************************************************************************

Renders a dot graph for a simple tree, with blackboxes.

    Generate Full Dot Graph

        py-trees-demo-dot-graphs

    With Varying Visibility Levels

        py-trees-demo-dot-graphs --level=all
        py-trees-demo-dot-graphs --level=detail
        py-trees-demo-dot-graphs --level=component
        py-trees-demo-dot-graphs --level=big_picture

*******************************************************************************

Writing /mnt/mervin/workspaces/devel/py_trees/src/py_trees/py_trees/demo_dot_graphs_fine_detail.dot
Traceback (most recent call last):
  File "/mnt/mervin/workspaces/devel/py_trees/src/py_trees/.venv/bin/py-trees-demo-dot-graphs", line 11, in <module>
    load_entry_point('py-trees', 'console_scripts', 'py-trees-demo-dot-graphs')()
  File "/mnt/mervin/workspaces/devel/py_trees/src/py_trees/py_trees/demos/dot_graphs.py", line 121, in main
    py_trees.display.render_dot_tree(root, args.enum_level)
  File "/mnt/mervin/workspaces/devel/py_trees/src/py_trees/py_trees/display.py", line 687, in render_dot_tree
    writer(pathname)
  File "/mnt/mervin/workspaces/devel/py_trees/src/py_trees/.venv/lib/python3.8/site-packages/pydot.py", line 1826, in write
    f.write(s)
  File "/usr/lib/python3.8/encodings/iso8859_15.py", line 19, in encode
    return codecs.charmap_encode(input,self.errors,encoding_table)[0]
UnicodeEncodeError: 'charmap' codec can't encode character '\u24c2' in position 344: character maps to <undefined>
```

--- a/py_trees/console.py
+++ b/py_trees/console.py
@@ -330,6 +330,15 @@ def logfatal(message):
 ##############################################################################
 
 if __name__ == '__main__':
+
+    # To test without unicode, configure a non utf-8 locale:
+    #
+    #   $ cat /usr/share/i18n/SUPPORTED | grep en_US
+    #   $ sudo locale-gen en_US.ISO-8859-15
+    #   $ sudo update-locale
+    #   $  locale -a
+    #   $ python3 ./console.py
+
     for colour in colours:
         pretty_print("dude\n", colour)
     logdebug("loginfo message")
@@ -339,7 +348,8 @@ if __name__ == '__main__':
     pretty_print("red\n", red)
     print("some normal text")
     print(cyan + "    Name" + reset + ": " + yellow + "Dude" + reset)
-    print("special characters are\n")
+    print(f"Has Unicode: {has_unicode()}")
+    print("Unicode Characters:\n")
     print("lightning_bolt: {}".format(lightning_bolt))
     print("double_vertical_line: {}".format(double_vertical_line))
     print("check_mark: {}".format(check_mark))
@@ -347,4 +357,3 @@ if __name__ == '__main__':
     print("left_arrow: {}".format(left_arrow))
     print("right_arrow: {}".format(right_arrow))
     print("circled_m: {}".format(circled_m))
-    # print("has unicode: {}".format(has_unicode()))

--- a/py_trees/display.py
+++ b/py_trees/display.py
@@ -47,6 +47,7 @@ unicode_symbols = {
     'bold': console.bold,
     'bold_reset': console.reset,
     'memory': console.circled_m,
+    'synchronised': console.lightning_bolt,
     'sequence_with_memory': u'{-}',
     'selector_with_memory': u'{o}',
     composites.Sequence: u'[-]',
@@ -69,6 +70,7 @@ ascii_symbols = {
     'bold': console.bold,
     'bold_reset': console.reset,
     'memory': 'M',
+    'synchronised': 's',
     'sequence_with_memory': '{-}',
     'selector_with_memory': '{o}',
     composites.Sequence: "[-]",
@@ -91,6 +93,7 @@ xhtml_symbols = {
     'bold': '<b>',
     'bold_reset': '</b>',
     'memory': '<text>&#x24C2;</text>',
+    'synchronised': '<text>&#9889;</text>',
     'sequence_with_memory': '<text>{-}</text>',
     'selector_with_memory': '<text>{o}</text>',
     composites.Sequence: '<text>[-]</text>',
@@ -452,15 +455,16 @@ def dot_tree(
         # for their derived composites.
         prefix = ""
         policy = ""
+        symbols = unicode_symbols if console.has_unicode() else ascii_symbols
         if isinstance(behaviour, composites.Composite):
             try:
                 if behaviour.memory:
-                    prefix += console.circled_m
+                    prefix += symbols["memory"]  # console.circled_m
             except AttributeError:
                 pass
             try:
                 if behaviour.policy.synchronise:
-                    prefix += console.lightning_bolt
+                    prefix += symbols["synchronised"]  # console.lightning_bolt
             except AttributeError:
                 pass
             try:

